src/System.Net.Http/src/System/Net/Http/HttpClientHandler.Unix.cs

```diff
@@ -160,14 +161,15 @@ protected override void Dispose(bool disposing)
 
         protected internal override Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
         {
-            return _curlHandler.SendAsync(request, cancellationToken);
+            return _curlHandlerPipeline.SendAsync(request, cancellationToken);
```

Comment:
It kinda looks weird that to send the request, you would use a DiagnosticsHandler. However, I don't have better suggestion.